src/azure-cli/azure/cli/command_modules/acr/taskrun.py

```diff
@@ -0,0 +1,41 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+from ._utils import (
+    validate_managed_registry
+)
+
+
+TASKRUN_NOT_SUPPORTED = 'TaskRun is only supported for managed registries.'
+
+
+def acr_taskrun_show(cmd,
+                     client,
+                     taskrun_name,
+                     registry_name,
+                     resource_group_name=None):
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASKRUN_NOT_SUPPORTED)
+    return client.get(resource_group_name, registry_name, taskrun_name)
+
+
+def acr_taskrun_list(cmd,
+                     client,
+                     registry_name,
+                     resource_group_name=None):
+
+    _, resource_group_name = validate_managed_registry(
+        cmd, registry_name, resource_group_name, TASKRUN_NOT_SUPPORTED)
+    return client.list(resource_group_name, registry_name)
+
+
+def acr_taskrun_delete(cmd,
+                       client,
+                       taskrun_name,
+                       registry_name,
+                       resource_group_name=None):
```

Comment:
can you add `yes=False` option to confirm the delete operation? You can check acr_repository_delete.

Reply:
I notice acr task delete also need to add the same confirmation.

---
In reply to: [368712257](https://github.com/Azure/azure-cli/pull/11858#discussion_r368712257) [](ancestors = 368712257)